modulus/models/diffusion/preconditioning.py

```diff
@@ -780,10 +777,14 @@ def forward(
         c_in = 1 / (self.sigma_data**2 + sigma**2).sqrt()
         c_noise = sigma.log() / 4
 
-        x = torch.cat((x, img_lr), dim=1)
+        if img_lr is None:
+            arg = c_in * x
+        else:
+            img_lr = img_lr.to(torch.float32)
+            arg = torch.cat([c_in * x, img_lr], dim=1)
 
         F_x = self.model(
-            (c_in * x).to(dtype),
+            (c_in * arg).to(dtype),
```

Comment:
this code is still doing `c_in * c_in * x`. this should fix:

```suggestion
            arg,
```